### PR TITLE
Microsoft Azure StorageCache new API version

### PR DESCRIPTION
**Pull request by romahamu**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2019-11-01/storagecache.json [hunk 1]
```diff
@@ -0,0 +1,1592 @@
...
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "tags": [
+          "Caches"
+        ],
+        "description": "Create or update a Cache.",
+        "operationId": "Caches_CreateOrUpdate"
+      },
+      "patch": {
```
Comment (yungezz) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2019-11-01/storagecache.json]: what's difference between this patch and above put on updating behavior?

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2019-11-01/storagecache.json [hunk 2]
```diff
@@ -0,0 +1,1592 @@
...
+        },
+        "pendingFirmwareVersion": {
+          "description": "When firmwareUpdateAvailable is true, this field holds the version string for the update.",
+          "type": "string",
+          "readOnly": true
+        }
+      },
+      "type": "object"
+    },
+    "UnknownProperties": {
```
Comment (yungezz) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2019-11-01/storagecache.json]: not sure this is right way to define/naming additional property for extension, need ARM opinion

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2019-11-01/storagecache.json (hunk 1 above)
Comment (omzevall) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2019-11-01/storagecache.json]: There is no real difference between PUT/PATCH in terms of updating behavior. Currently only tags can be updated on Caches.

Our understanding is that we needed PATCH to support updating tags as an Azure/ARM requirement.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2019-11-01/storagecache.json (hunk 2 above)
Comment (waffe7) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2019-11-01/storagecache.json]: This is just used in the UnknownTarget, and exists to support new ST types in API versions that existed before the ST type was introduced.

**Review by ChenTanyi — COMMENTED** [review]
I think the PR title should be changed, since people don't know what resource was added new api from this title.

File: specification/storagecache/resource-manager/readme.md [hunk 3]
```diff
@@ -26,7 +26,18 @@ These are the global settings for the StorageCache API.
 
 ``` yaml
 openapi-type: arm
-tag: package-2019-08
+tag: package-storagecache-2019-11-01
```
[the remark this answers is not in the document]
Comment (romahamu) [reply specification/storagecache/resource-manager/readme.md]: Fixed

File: specification/storagecache/resource-manager/readme.ruby.md [hunk 4]
```diff
@@ -8,6 +8,24 @@ package-version: 2019-08
...
+  - tag: package-2019-08
+```
+
+### Tag: package-2019-11-01 and ruby
+
+These settings apply only when `--tag=package-storagecache-2019-11-01 --ruby` is specified on the command line.
+Please also specify `--ruby-sdks-folder=<path to the root directory of your azure-sdk-for-ruby clone>`.
+
+``` yaml $(tag) == 'package-storagecache-2019-11-01' && $(ruby)
+namespace: "Azure::Redis::Mgmt::V2019_11_01"
```
[the remark this answers is not in the document]
Comment (romahamu) [reply specification/storagecache/resource-manager/readme.ruby.md]: Fixed

**Review by KrisBash — APPROVED** [review]
LGTM

**Review by ChenTanyi — APPROVED** [review]

**Pull request merged**